PowerEditor/src/Utf8_16.cpp

```diff
@@ -445,13 +454,19 @@ size_t Utf8_16_Write::convert(char* p, size_t _size)
 			{
                 // Write the BOM
 				m_pNewBuf = (ubyte*)new ubyte[sizeof(utf16) * (_size + 1)];
-                memcpy(m_pNewBuf, k_Boms[m_eEncoding], 2);
-	            pCur = (utf16*)&m_pNewBuf[2];
+				if (m_pNewBuf)
+				{
+					memcpy(m_pNewBuf, k_Boms[m_eEncoding], 2);
+					pCur = (utf16*)&m_pNewBuf[2];
+				}
             }
 			else
 			{
 				m_pNewBuf = (ubyte*)new ubyte[sizeof(utf16) * _size];
-	            pCur = (utf16*)m_pNewBuf;
+				if (m_pNewBuf)
+				{
+					pCur = (utf16*)m_pNewBuf;
+				}

```

Comment:
It's good to check if the allocation has been done correctly.
However, in the case of failure of allocation, pCur will be NULL.
And the code below will use pCur as NULL, crash will still happen.

IMO, it would be better to throw `std::bad_alloc` as m_pNewBuf get NULL from ***new*** operator.

Reply:
> I would go for the try catch with std::bad_alloc for Utf8_16_Write and the classic nullpointer check within Utf8_16_Read. So we get no problem with FileManager::loadFileData() where __try is used?

Yes, I think it's a reasonable solution.

> What do you think about the value of m_nBufSize in Utf8_16_Write::convert()?
I think it should be 0 in case the buffer gets deleted and in case a new alloc fails.

As well, it seems 0 is a good value of m_nBufSize to me while alloc fails.